prac_07/project_management.py

```diff
@@ -0,0 +1,159 @@
+                          project_completion_percentage)
+        projects.append(project)
+    in_file.close()
+    return projects
+
+
+def save_projects(filename, projects):
+    """Save list of project objects to chosen file. """
+    out_file = open(filename, 'w')
+    out_file.writelines("Name\tStart Date\tPriority\tCost Estimate\tCompletion Percentage\n")
+    for project in projects:
+        print(project.name, project.start_date, project.priority, project.cost_estimate, project.completion_percentage,
+              sep="\t", file=out_file)
+    out_file.close()
+
+
+def display_projects(projects):
+    """Display projects listed in completed and incomplete projects. """
+    completed_projects = []
+    incomplete_projects = []
+    for project in projects:
+        if project.completion_percentage == 100:
+            completed_projects.append(project)
+        else:
+            incomplete_projects.append(project)
+    completed_projects.sort()
+    incomplete_projects.sort()
+
+    print("Incomplete Projects: ")
+    for project in incomplete_projects:
+        print(f"\t{project}")
+    print("Completed Projects: ")
+    for project in completed_projects:
+        print(f"\t{project}")
+
+
+def filter_projects(projects):
+    """Filter projects by date, and display sorted by date. """
+    # date_string = input("Date (d/m/yyyy): ")
+    # date = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
+    filter_date = input("Date (d/m/yyyy): ")
+    for project in projects:
+        projects.sort()
+        if project.start_date > filter_date:
+            print(project)
+
+
+def add_project(projects):
+    """Add new project with details"""
+    # new_project = []
+    print("Let's add a new project")
+    new_name = input("Name: ")
+    # new_start_date_string = input("Start date (d/m/yyyy): ")
+    # new_start_date = datetime.datetime.strptime(new_start_date_string, "%d/%m/%Y").date()
+    new_start_date = input("Start date (d/m/yyyy): ")
+    new_priority = int(input("Priority: "))
+    new_cost_estimate = float(input("Cost estimate: "))
+    new_percent_complete = int(input("Percent complete: "))
+    print(f"Project: {new_name}, start: {new_start_date}, priority {new_priority}, "
+          f"estimate: ${new_cost_estimate:,.2f}, completion: {new_percent_complete}%, Added to project list. ")
+    # new_project.append(Project(new_name, new_start_date, new_priority, new_cost_estimate, new_percent_complete))
+    projects.append(Project(new_name, new_start_date, new_priority, new_cost_estimate, new_percent_complete))
+    return projects
+
+
+def update_project(projects):
+    """Update project percentage and/or priority, leave blank to retain info. """
+    number_of_projects = 0
+    for project in projects:
+        number_of_projects += 1
+        print(f"{number_of_projects} {project}")
+
+    project_choice = int(input("Project choice: "))
+    project_index = project_choice - 1
+    print(projects[project_index])
```

Comment:
The sample output doesn't ask for this but I think it makes more sense the first project being 1. rather than 0.